### PR TITLE
build: run code coverage on min and max py vers

### PR DESCRIPTION
**Pull request by peterschutt**
We do a lot of `pragma: no cover` due to coverage only being reported on the highest version of python.

This PR modifies coverage reporting so that we run coverage tests on the lowest and highest supported versions, and combines the coverage reports.

### Pull Request Checklist

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [ ] Pre-Commit Checks were ran and passed
- [ ] Tests were ran and passed

### Description

-

### Close Issue(s)

-

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -191,7 +191,7 @@ exclude_lines = [
   'except ImportError as e:',
   'except ImportError:',
```
Comment (sobolevn) [inline pyproject.toml:192]:
```suggestion
  'except ImportError.*',
```

Comment (sobolevn) [reply pyproject.toml:192]: Also, consider using https://github.com/asottile/covdefaults
It also has conditional coverage.

**Review by provinzkraut — APPROVED** [review]
Great stuff!

File: pyproject.toml (hunk 1 above)
Comment (peterschutt) [reply pyproject.toml]: > Also, consider using https://github.com/asottile/covdefaults It also has conditional coverage.

Thanks for the suggestion. I'll experiment with adopting it in a follow up.

**Pull request merged**